### PR TITLE
Properly set the Origin header for WebSocket requests

### PR DESCRIPTION
**Pull request by nox**
***
[Preview](https://s3.amazonaws.com/pr-preview/nox/fetch/websocket-origin.html) | [Diff](https://s3.amazonaws.com/pr-preview/whatwg/fetch/cbca2c2...nox:c43ea5c.html)

### CODE REVIEW

File: fetch.bs [hunk 1]
```diff
@@ -3247,7 +3247,8 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
           concept-as-bytes -->
 
  <li><p>If the <i>CORS flag</i> is set or <var>httpRequest</var>'s <a for=request>method</a> is
```
Comment (annevk) [inline fetch.bs]: s/ or/,/

File: fetch.bs [hunk 2]
```diff
@@ -3247,7 +3247,8 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
...
- neither `<code>GET</code>` nor `<code>HEAD</code>`, then <a for="header list">append</a>
+ neither `<code>GET</code>` nor `<code>HEAD</code>` or <var>httpRequest</var>'s
```
Comment (annevk) [inline fetch.bs]: s/ or/, or/

File: fetch.bs [hunk 3]
```diff
@@ -3247,7 +3247,8 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
...
+ <a for=request>mode</a> is <code>"websocket"</code>, then <a for="header list">append</a>
```
Comment (annevk) [inline fetch.bs]: Double quotes go on the outside.

**Comment by zcorpan** [comment]
./websockets/opening-handshake/003.html
./websockets/opening-handshake/003-sets-origin.worker.js

I actually think pywebsocket would refuse the connection if the `Origin` header was missing, which would fail lots of websocket tests.

**Pull request merged**